ngraph/frontend/onnx_import/src/editor/editor.cpp

```diff
@@ -142,6 +178,60 @@ namespace
             *(tensor_type->mutable_shape()) = std::move(new_onnx_shape);
         }
     }
+
+    void modify_initializer(TensorProto& initializer,
+                            const std::string& name,
+                            const std::shared_ptr<ngraph::op::Constant> values,
+                            ValueInfoProto* input)
+    {
+        const auto elem_type = values->get_element_type();
+        if (NG_2_ONNX_TYPES.count(elem_type) == 0)
+        {
+            throw ngraph_error("Initializer '" + name + "' type cannot be set to: " +
+                               element::Type(elem_type).get_type_name() +
+                               ". This type is not allowed in ONNX.");
+        }
+
+        initializer.Clear();
+
+        *initializer.mutable_name() = name;
+
+        initializer.set_data_type(NG_2_ONNX_TYPES.at(values->get_element_type()));
+
+        for (const auto& dim : values->get_shape())
+        {
+            initializer.add_dims(dim);
+        }
+
+        const auto data_size_in_bytes =
+            shape_size(values->get_shape()) * get_onnx_data_size(initializer.data_type());
+
+        initializer.set_raw_data(values->get_data_ptr(), data_size_in_bytes);
+        /* It's a workaround for problem with multitle instances of fixed_address_empty_string in
+         * protobuf lib caused by dynamic linking.
+         * The problem can be observed as two tensors pointing the same data.
+         */
+        initializer.mutable_raw_data();
+
+        // update input with type and shape of initializer
+        if (input && input->has_type())
+        {
+            auto type_proto = input->mutable_type();
+            if (type_proto->has_tensor_type())
+            {
+                auto tensor_type = type_proto->mutable_tensor_type();
+
+                auto shape = tensor_type->mutable_shape();
+                shape->clear_dim();
+                for (size_t i = 0; i < initializer.dims_size(); ++i)
+                {
+                    shape->add_dim()->set_dim_value(initializer.dims(i));
+                }
+
+                tensor_type->set_elem_type(initializer.data_type());
+            }
+        }
```

Comment:
newly created initializer should have the same shape as targeted input, is it correct?
why do we need to update it?

Reply:
_is it correct?_
No, user is allowed to use different type or shape, therefore (if applicable) input is changed accordingly.
_why do we need to update it?_
To keep these tensors aligned.